bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java

```diff
@@ -114,6 +114,9 @@ public void run() {
                 }
                 if (lastBridgeConnectionState) {
                     doConnectedRun();
+                    if (thing.getStatus() != ThingStatus.ONLINE) {
+                        updateStatus(ThingStatus.ONLINE);
+                    }
```

Comment:
Let the framework handle the checking for you.
```suggestion
                    updateStatus(ThingStatus.ONLINE);
```

Reply:
I checked the framework code and each call to updateStatus will produce one event on the event bus (even 2 in case of status change).
With the sensor polling run by default twice per second and possibly every 50 ms, this could lead to a massive event publishing on the bus. My test is here to avoid that.

Reply:
I realize that we could have a massive number of OFFLINE events on the bus too. I think the same check should be applied to updateStatus(OFFLINE).
This case is of course less common.

Reply:
I think in general that this is something that the framework should have been optimized for since bindings shouldn't be expected to perform this check themselves.

After looking up the core code it does seem that it would publish to the event bus every time you call that, but it wouldn't publish the additional change event if it hasn't changed.

https://github.com/openhab/openhab-core/blob/61e17ce39e83daa7f1d8c41512dead8d95d6e5d9/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java#L1147

So I guess you might have a point in trying to prevent that mandatory event publish. Seems weird that it would do that but we can't really change the core atm and your check is harmless so just leave your status check in place.

Reply:
> I realize that we could have a massive number of OFFLINE events on the bus too. I think the same check should be applied to updateStatus(OFFLINE).
This case is of course less common.

I don't think you should check in this case since it doesn't occur as often.